ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -561,7 +589,9 @@ runPure cfg = \case
     FollowerInstruction flr  -> err MbChainUpdate       $ updateE (Model.followerInstruction flr allComponents)
     FollowerForward flr pts  -> err MbPoint             $ updateE (Model.followerForward flr pts)
     FollowerClose flr        -> ok  Unit                $ update_ (Model.followerClose flr)
-    RunBgTasks               -> ok  Unit                $ update_ (Model.garbageCollect k . Model.copyToImmutableDB k)
+    PersistBlks              -> ok  Unit                $ update_ (Model.copyToImmutableDB k False)
+    PersistBlksThenGC        -> ok  Unit                $ update_ (Model.copyToImmutableDB k True)
+    UpdateLedgerSnapshots    -> ok  Unit                $ ((), ) -- The model does not capture the notion of ledger snapshots.
```

Comment:
(For the record, we talked about this during our Sync call. Considered some options. Damian will explore and judge.)

Reply:
I think this is the only remaining issue. I look forward to chatting about it with you. We can do it in our Tuesday call if not on Monday via Slack.

Reply:
OK. We talked it through, and here's my summary of what we concluded.

(Please object if you think I get anything wrong here. Otherwise please mark this comment Thread as Resolved.)

- Right now, for `UpdateLedgerSnapshots`, the model merely checks if the database is open or not.
- So, while the SUT performs the actual ledger snapshot update, the model doesn't confirm anything about it except whether it should have thrown `ClosedDBError`.
- This is fine for this PR.
- We don't want to remove `UpdateLedgerSnapshots` altogether because a) the real node takes snapshots sometimes and b) taking a snapshot should not interfere with any of the other commands, which the current model is scrutinizing and therefore would notice. (~"frame rule"/confirming it's a "no-op" externally)
- We should open a separate follow-up tech-debt Issue to enrich the model so that does scrutinize the state of the ledger snapshots in some way. EG the model could predict the slotnos of the snapshots at any given time -- eg we'd have to add a `intListAvailableSnapshots` command to the (testing) API of `ChainDB`). But it'd be nice if the model was also testing the snapshotting semantics, to some degree more than "when should the command throw `ClosedDBError`".

Reply:
That sounds good to me. I'll update the comment as discussed on slack, have a team member reviewing it, and open a new PR referring to this line.

Reply:
I created the new issue, and I'll cross-link it with the comment in the code.